src/tests/functional/frame_navigation_tests.ts

```diff
@@ -23,6 +23,13 @@ test("test frame navigation with exterior link", async ({ page }) => {
   await nextEventOnTarget(page, "frame", "turbo:frame-load")
 })
 
+test("test frame navigation emits fetch-request-error event when offline", async ({ page }) => {
+  await page.goto("/src/tests/fixtures/tabs.html")
+  await page.context().setOffline(true)
+  await page.click("#tab-2")
+  await nextEventNamed(page, "turbo:fetch-request-error")
```

Comment:
@srt32 to use this helper method, you need to add an entry for the event name to the src/tests/test.js file.